monkey/infection_monkey/exploit/tools/brute_force_credentials_generator.py

```diff
@@ -0,0 +1,71 @@
+from itertools import chain
+from typing import Callable, Dict, Iterable, List, Sequence, Set, Type
+
+from common.credentials import Credentials, Identity, Secret
+from common.utils.code_utils import apply_filters
+
+
+def generate_brute_force_credentials(
+    input_credentials: Iterable[Credentials],
+    identity_filters: Iterable[Callable[[Identity], bool]] = (),
+    secret_filters: Iterable[Callable[[Secret], bool]] = (),
+) -> Sequence[Credentials]:
+    """
+    Generates all possible combinations of identities and secrets from the inputs
+
+    Given some Credentials objects, this function will generate all possible combinations of
+    identities and secrets. The function will preserve "complete" credentials (i.e. both identity
+    and secret are provided) and will put them first in the output sequence. This allows callers to
+    prioritize using known identity/secret pairs before attempting to use random combinations.
```

Comment:
I think this comment could be simplified by saying that "This method will return a sequence of credentials starting with input credentials and ending with credentials generated from input identities and secrets randomly"